### PR TITLE
Add documentation about the various tables in the DB

### PR DESCRIPTION
**Pull request by Geod24**

### CODE REVIEW

**Comment by Geod24** [comment]
https://github.com/Geod24/agora/blob/db-doc/doc/Database.md

File: doc/Database.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+| nonce           | TEXT     | Point            |                             | The `R` used to sign the `Enrollment`              |
+| active          | INTEGER  | EnrollmentStatus | PRIMARY KEY (with `key`)    | An enum (boolean). Should be removed.              |
+
+### `node_enroll_data` table
+
+This table should be removed in favor of querying `validator_set` directly.
+
+| Field name       | SQL Type | D type | Attributes  | Comment                                                           |
+|------------------|----------|--------|-------------|-------------------------------------------------------------------|
+| node_enroll_data | TEXT     | string | PRIMARY KEY | Used as an AA. Valid values: `utxo`, `random_seed`                |
```
Comment (linked0) [inline doc/Database.md]: The field name should be `key`, not `node_enroll_data`.

File: doc/Database.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+| data          | BLOB     | Transaction  | NOT NULL             | Binary-serialized `Channel` |
+| update_signer | BLOB     | UpdateSigner |                      | Doesn't seem to be used     |
+
+
+### `flash_metadata` table
+
+| Field name | SQL Type | D type | Attributes           | Comment                         |
+|------------|----------|--------|----------------------|---------------------------------|
+| meta       | BLOB     | Hash   | PRIMARY KEY NOT NULL | Only ever set to `1` if present |
+| data       | BLOB     |        | NOT NULL             | Binary-serialized `mixin` ?     |
```
Comment (linked0) [inline doc/Database.md]: What does the `?` mean?

**Review by linked0 — APPROVED** [review]
LGTM with two requests.

File: doc/Database.md (hunk 2 above)
Comment (Geod24) [reply doc/Database.md]: Means I am not quite sure. CC @AndrejMitrovic .

Comment (AndrejMitrovic) [reply doc/Database.md]: Yes it's binary. I don't know why we should document these? It's going to get outdated anyway.

Comment (Geod24) [reply doc/Database.md]: > I don't know why we should document these?

I'm trying to get things under a state in which we  can actually get an overview of the state of a node by looking at the sqlite3 DBs. The first step was documenting everything so we'd know what tables there are and what they are used for. I felt it was a pre-requisite to make the ORM issue feasible.

File: doc/Database.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+This contains Flash-related data.
+
+### `channels` table
+
+| Field name    | SQL Type | D type       | Attributes           | Comment                     |
+|---------------|----------|--------------|----------------------|-----------------------------|
+| chan_id       | BLOB     | Hash         | PRIMARY KEY NOT NULL | Binary-serialized hash      |
+| data          | BLOB     | Transaction  | NOT NULL             | Binary-serialized `Channel` |
+| update_signer | BLOB     | UpdateSigner |                      | Doesn't seem to be used     |
```
Comment (Geod24) [inline doc/Database.md]: Is this comment right @AndrejMitrovic ?

File: doc/Database.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+
+### `flash_metadata` table
+
+| Field name | SQL Type | D type | Attributes           | Comment                         |
+|------------|----------|--------|----------------------|---------------------------------|
+| meta       | BLOB     | Hash   | PRIMARY KEY NOT NULL | Only ever set to `1` if present |
```
Comment (Geod24) [inline doc/Database.md:104]: Ditto, not sure I understood this correctly @AndrejMitrovic @omerfirmak ?

File: doc/Database.md (hunk 3 above)
Comment (AndrejMitrovic) [reply doc/Database.md]: It's used, just mark it as binary for now.

**Pull request merged**